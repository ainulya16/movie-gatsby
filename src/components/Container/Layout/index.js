import React from 'react';
import PropTypes from 'prop-types';
import { Layout } from 'antd';

import Header from '../../Header';
import Footer from './Footer';
import ContainerWrapper from '../../Wrapper/Container';

class Container extends React.PureComponent {
  render() {
    const { children, custom } = this.props;
    return (
      <Layout>
        <Header />
        <Layout>
          <ContainerWrapper>{children}</ContainerWrapper>
          <Footer custom={custom} />
        </Layout>
      </Layout>
    );
  }
}

Container.propTypes = {
  children: PropTypes.node.isRequired,
  custom: PropTypes.bool,
};

Container.defaultProps = {
  custom: false,
};

export default Container;
