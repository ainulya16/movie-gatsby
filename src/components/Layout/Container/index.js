import styled from 'styled-components';

const Container = styled.div`
  display: flex;
  flex-direction: column;
  min-height: 100%;
  .flex-center {
    display: flex;
    flex: 1;
    justify-content: center;
    align-items: center;
  }
`;

export default Container;
