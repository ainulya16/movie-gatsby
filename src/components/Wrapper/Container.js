import styled from 'styled-components';

const Container = styled.div`
  @media (max-width: 768px) {
    margin: 5px;
  }
  /* Small devices (tablets, 768px and up) */
  @media (min-width: 768px) and (max-width: 992px) {
    margin: 10px;
  }
  /* Medium devices (desktops, 992px and up) */
  @media (min-width: 992px) and (max-width: 1200px) {
    margin: 44px;
  }
  /* Large devices (large desktops, 1200px and up) */
  @media (min-width: 1200px) {
    margin: 64px;
  }

  /* padding: 0 50px; */
  display: block;
  margin: 64px;
  @media (max-width: 768px) {
    margin: 64px 0 0 0;
  }
`;

export default Container;
