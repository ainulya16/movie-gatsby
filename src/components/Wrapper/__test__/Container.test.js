import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Container from '../Container';

describe('Container', () => {
  it('should be exists', () => {
    const wrapper = shallow(<Container />);
    expect(wrapper.exists()).toBe(true);
  });
  it('should match the snapshot', () => {
    const tree = renderer.create(<Container />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
