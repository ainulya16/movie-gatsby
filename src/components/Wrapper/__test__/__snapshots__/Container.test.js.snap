// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container should match the snapshot 1`] = `
.c0 {
  display: block;
  margin: 64px;
}

@media (max-width:768px) {
  .c0 {
    margin: 5px;
  }
}

@media (min-width:768px) and (max-width:992px) {
  .c0 {
    margin: 10px;
  }
}

@media (min-width:992px) and (max-width:1200px) {
  .c0 {
    margin: 44px;
  }
}

@media (min-width:1200px) {
  .c0 {
    margin: 64px;
  }
}

@media (max-width:768px) {
  .c0 {
    margin: 64px 0 0 0;
  }
}

<div
  className="c0"
/>
`;
